src/virtio/mmio.c

```diff
@@ -204,6 +204,7 @@ static bool handle_virtio_mmio_reg_write(virtio_device_t *dev, size_t vcpu_id, s
             }
             break;
         case REG_RANGE(REG_VIRTIO_MMIO_QUEUE_NOTIFY, REG_VIRTIO_MMIO_INTERRUPT_STATUS):
+            dev->data.QueueNotify = (uint32_t)data;
```

Comment:
If I'm understanding correctly we should be able to do this:

```suggestion
            /* We do not support  VIRTIO_F_NOTIFICATION_DATA so the data will always be the queue index. */
            dev->data.QueueNotify = (uint32_t)data;
```

Reply:
It would be better if we could assert ` VIRTIO_F_NOTIFICATION_DATA` is not negotiated, I cannot remember if this is information we can get from the MMIO implementation.

Reply:
Negotiation happens in the actual driver implementations with `get_device_features` and `set_driver_features`. We could either trust devices to not set `VIRTIO_F_NOTIFICATION_DATA` or check / mask. the result of `get_device_features`
```c
case REG_RANGE(REG_VIRTIO_MMIO_DEVICE_FEATURES, REG_VIRTIO_MMIO_DEVICE_FEATURES_SEL):
    success = dev->funs->get_device_features(dev, &reg);
    reg &= ~BIT_HIGH(VIRTIO_F_NOTIFICATION_DATA);
    // could mask out other unsupported features...
    break;
```